### ISSUE TITLE
IDE finds no issues with variables1

### ISSUE THREAD

**Issue by Konglomneshued** [rust-lang/rustlings#1323]
<details>
<summary>I have syntax hightlighting on normal Rust project.</summary>
<img src="https://user-images.githubusercontent.com/94726665/210679086-f041ccd9-728e-415c-855e-7b10fb7d94c3.png" alt="" />
</details>

But when I open the rustlings project I have none.

<details>
<summary>Fleet:</summary>
<img src="https://user-images.githubusercontent.com/94726665/210678870-a14fd243-37ca-4a50-bde0-e84e9c2fc3b8.png" alt="" />
</details>

<details>
<summary>WebStorm:</summary>
<img src="https://user-images.githubusercontent.com/94726665/210678872-acabff57-ac24-40cb-ad5e-aca4cb8524af.png" alt="" />
</details>

<details>
<summary>VS Code:</summary>
<img src="https://user-images.githubusercontent.com/94726665/210678874-4b20fd3f-b2b6-45ce-aac1-6ddaac015665.png" alt="" />
</details>

**Comment by diannasoreil** [comment]
That's probably because you don't have rust-analyzer support enabled – run `rustlings lsp` once to get that set up. See: https://github.com/rust-lang/rustlings#enabling-rust-analyzer

**Comment by Konglomneshued** [comment]
> That's probably because you don't have rust-analyzer support enabled – run `rustlings lsp` once to get that set up. See: https://github.com/rust-lang/rustlings#enabling-rust-analyzer

No, I had already enabled it.  Besides, WebStorm does not use rust-analyzer.

**Comment by diannasoreil** [comment]
Not sure then, I've heard from people that JetBrains IDEs _should_ just work. Are you using the latest version or are you on `main`?

**Comment by orblazer** [comment]
Hello, i have exactly same issue.
First i have try to downgrade `rust-analyzer` to `v0.3.1451`, try to set `root_module` to absolute paths (like i could found on #1462), but even with that that still doesn't work...

The only solution i have found is to delete `rust-project.json` and using the code i have could found here https://github.com/rust-lang/rustlings/issues/1421#issuecomment-1483773498 for generate `mod.rs` and now every look to work correctly.

**Comment by shadows-withal** [comment]
This may be related to the same issue that the people in #1462 were having, @orblazer, by chance, are you using Windows?

**Comment by orblazer** [comment]
@shadows-withal no i use Manjaro with `6.1.26-1-MANJARO` kernel version.

**Comment by Konglomneshued** [comment]
> This may be related to the same issue that the people in #1462 were having, @orblazer, by chance, are you using Windows?

I don't remember, sorry.  It's three months later and I switched operating systems somewhere down the line.  It's possible I was, or I was using Linux.

**Comment by shadows-withal** [comment]
Hm, actually not related to that other issue, it seems.

**Comment by mo8it** [comment]
This should be resolved when https://github.com/rust-lang/rustlings/issues/1935 is closed.

**Comment by jsejcksn** [comment]
> This should be resolved when #1935 is closed.

@mo8it I just tested the [`v6` branch](https://github.com/rust-lang/rustlings/tree/v6) (specifically https://github.com/rust-lang/rustlings/commit/8c8f30d8ce3b732de649938d8945496bd769ac22) and see the same behavior as OP with rust-analyzer in VS Code.

Version data:
```
% rustc --version
rustc 1.77.2 (25ef9e3d8 2024-04-09)

% rustlings --version
rustlings 6.0.0

% code --version
1.88.1
e170252f762678dec6ca2cc69aba1570769a5d39
arm64

% code --list-extensions --show-versions | grep 'rust-analyzer'
rust-lang.rust-analyzer@0.3.1916
```

**Comment by mo8it** [comment]
I am sorry for the confusion, but the v6 branch is not really functional yet 😅

As I mentioned in https://github.com/rust-lang/rustlings/issues/1919#issuecomment-2050523778, I am still working on the code for the next couple of days.
